src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3150,6 +3150,18 @@ internal bool ShouldSuppressNullableAnnotations(Symbol definition)
             return true;
         }
 
+        private bool HasAllowedNullableAssemblyOptOut(AssemblySymbol assembly)
+        {
+            return ((SourceModuleSymbol)SourceModule).IsNullableOptOutForAssembly(assembly) &&
+                (this.GetNullableReferenceFlags() & NullableReferenceFlags.AllowAssemblyOptOut) != 0;
+        }
+
+        private bool HasAllowedNullableOptOut(Symbol definition)
```

Comment:
>HasAllowedNullableOptOut [](start = 21, length = 24)

Similar here. "HaveNullableOptOutForDefinition"? Consider also adding an assert that the symbol is not an assembly.